### ISSUE TITLE
Reduce ARC usage of the IO cache...

### ISSUE THREAD

**Issue by sean-** [joyent/pg_prefaulter#15]
While writing this I defaulted to using string slices as the arguments, which is efficient, but not necessarily ideal.

The current `_IOCacheKey` is too large:

```go
type _IOCacheKey struct {
        Tablespace string
        Database   string
        Relation   string
        Block      string
}

_IOCacheKey.Tablespace string: 0-16 (size 16, align 8)
_IOCacheKey.Database string: 16-32 (size 16, align 8)
_IOCacheKey.Relation string: 32-48 (size 16, align 8)
_IOCacheKey.Block string: 48-64 (size 16, align 8)
```

Plus the size of byte array backing the `string`.  It could/should be:

```go
// type'ed uint64's
type _IOCacheKey struct {
        Tablespace uint64
        Database   uint64
        Relation   uint64
        Block      uint64
}

_IOCacheKey.Tablespace string: 0-8 (size 8, align 8)
_IOCacheKey.Database string: 8-16 (size 8, align 8)
_IOCacheKey.Relation string: 16-24 (size 8, align 8)
_IOCacheKey.Block string: 24-32 (size 8, align 8)
```

This change will incur a parsing overhead but we have CPU to burn in the common case and are memory limited, not CPU limited.  The current CPU overhead of `pg_prefaulter` running on a busy follower is between 1-2% CPU, so we have CPU budget to spare.

**Comment by sean-** [comment]
Extra reference material for interested folks:

* https://github.com/dominikh/go-tools/tree/master/cmd/structlayout
* https://blog.golang.org/slices
* https://blog.golang.org/go-slices-usage-and-internals

**Comment by sean-** [comment]
Additional testing is required to verify this, but this should have been closed with 5250762 when the following was committed:

https://github.com/joyent/pg_prefaulter/blob/5250762/agent/structs/cache.go#L18-L28

**Comment by sean-** [comment]
Small late measurement for posterity: resident size is now only `52M` down from ~`1G` previously.
